tests/bot/cogs/sync/test_cog.py

```diff
@@ -90,7 +91,6 @@ def test_sync_cog_init(self, sync_guild):
         sync_guild.assert_called_once_with()
         self.bot.loop.create_task.assert_called_once_with(mock_sync_guild_coro)
```

Comment:
I wonder... is there still no better way to do this despite all the async support that 3.8 had added?

Reply:
I'm not quite sure. The `create_task` method is not a coroutine but a regular method that, if actually used, returns an awaitable `Task` object. The thing I've changed here is that instead of mocking the `sync_guild` with an `AsyncMock`, I've mocked it with a `MagicMock`; the reason is that we're not actually interested in the async interface, but merely want to check if `create_task` was called with the right thing.

The latter is more difficult with an AsyncMock as when we call an AsyncMock, we get coroutine object back (we're not awaiting it anywhere here) and each time we call it, we get a different coroutine object back. That means that if were to go the route of `AsyncMock`, the call within the "actual" method in the cog will return a different coroutine object than the one we can create in the test method and we can't assert that `create_task` is called with the right argument: The coroutine objects are different.

By using a MagicMock instead, we don't need to deal with this more complicated coroutine object behavior as MagicMock will just return the same return value each time it's called, making this an easy assertion.

Reply:
Thanks for your input. I was never too happy with my solution for this but it was far cleaner than my first attempts.